### PR TITLE
Relative paths, jornalctl in additional logs

### PR DESCRIPTION
**Pull request by alpar-t**
This PR fixes the archive generation to preserve the paths relative to
the project directory.
It also fixes calling journalctl to get the system log.

### CODE REVIEW

**Comment by atorok** [comment]
@elasticmachine run 
elasticsearch-ci/packaging-sample-matrix

File: gradle/build-complete.gradle [hunk 1]
```diff
@@ -11,35 +11,31 @@ if (buildNumber) {
...
-                        commandLine 'bash', '-c', 'journalctl --since "1 hour ago" 2>&1 > journalctl.log'
-                    }
-                }
-                .onWindows {
-
-                }
-                .onMac {
-
-                }
+        if (OS.current() != OS.WINDOWS) {
```
Comment (mark-vieira) [inline gradle/build-complete.gradle]: I think we only want this to run on Linux since `journalctl` doesn't exist on OSX.

**Review by mark-vieira — APPROVED** [review]
One comment, otherwise LGTM 👍

**Comment by atorok** [comment]
@elasticmachine  run elasticsearch-ci/2

**Comment by atorok** [comment]
@elaticmachine run  elasticsearch-ci/2  and run elasticsearch-ci/packaging-sample-matrix

**Comment by imotov** [comment]
I got a couple of issues running this under elasticsearch-ci:

```
[2019-10-23T16:53:15.396Z] 
[2019-10-23T16:53:15.396Z] > Task :x-pack:plugin:integTestRunner FAILED
[2019-10-23T16:53:15.396Z] build complete, generating: /var/jenkins_home/jobs/elasticsearch-pipeline/workspace/elasticsearch/build/63.tar.bz2
[2019-10-23T16:53:15.396Z] bash: journalctl: command not found
[2019-10-23T16:53:15.396Z] Failed to archive additional logs
[2019-10-23T16:53:15.396Z] : Problem creating TAR: request to write '8192' bytes exceeds size in header of '6249522' bytes for entry 'docs/build/testclusters/integTest-0/logs/es.stdout.log'
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.ant.taskdefs.Tar.execute(Tar.java:351)
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
[2019-10-23T16:53:15.396Z] 	at jdk.internal.reflect.GeneratedMethodAccessor1799.invoke(Unknown Source)
[2019-10-23T16:53:15.396Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-10-23T16:53:15.396Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[2019-10-23T16:53:15.396Z] 	at groovy.util.AntBuilder.performTask(AntBuilder.java:338)
[2019-10-23T16:53:15.396Z] 	at groovy.util.AntBuilder.nodeCompleted(AntBuilder.java:283)
[2019-10-23T16:53:15.396Z] 	at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:80)
[2019-10-23T16:53:15.396Z] 	at groovy.util.BuilderSupport.doInvokeMethod(BuilderSupport.java:154)
[2019-10-23T16:53:15.396Z] 	at groovy.util.AntBuilder.doInvokeMethod(AntBuilder.java:218)
[2019-10-23T16:53:15.396Z] 	at org.gradle.api.internal.project.ant.BasicAntBuilder.doInvokeMethod(BasicAntBuilder.java:107)
[2019-10-23T16:53:15.396Z] 	at groovy.util.BuilderSupport.invokeMethod(BuilderSupport.java:67)
[2019-10-23T16:53:15.396Z] 	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:47)
[2019-10-23T16:53:15.396Z] 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:135)
[2019-10-23T16:53:15.396Z] 	at build_complete_6slzqvol7g9k9bb0j7zity5dc$_run_closure1.doCall(/var/jenkins_home/jobs/elasticsearch-pipeline/workspace/elasticsearch/gradle/build-complete.gradle:24)
[2019-10-23T16:53:15.396Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-10-23T16:53:15.396Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-10-23T16:53:15.396Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-10-23T16:53:15.396Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[2019-10-23T16:53:15.396Z] 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:104)
[2019-10-23T16:53:15.396Z] 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:326)
[2019-10-23T16:53:15.396Z] 	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:264)
[2019-10-23T16:53:15.396Z] 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041)
[2019-10-23T16:53:15.396Z] 	at groovy.lang.Closure.call(Closure.java:411)
[2019-10-23T16:53:15.396Z] 	at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:41)
[2019-10-23T16:53:15.396Z] 	at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:25)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:231)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:150)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:58)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:325)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:235)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:141)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:37)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
[2019-10-23T16:53:15.396Z] 	at com.sun.proxy.$Proxy15.buildFinished(Unknown Source)
[2019-10-23T16:53:15.396Z] 	at org.gradle.initialization.DefaultGradleLauncher.finishBuild(DefaultGradleLauncher.java:167)
[2019-10-23T16:53:15.396Z] 	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:129)
[2019-10-23T16:53:15.396Z] 	at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:106)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:60)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.invocation.GradleBuildController$1.execute(GradleBuildController.java:57)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:85)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.invocation.GradleBuildController$3.create(GradleBuildController.java:78)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:189)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.work.StopShieldingWorkerLeaseService.withLocks(StopShieldingWorkerLeaseService.java:40)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:78)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:57)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:63)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:39)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:51)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$3.call(RunAsBuildOperationBuildActionRunner.java:45)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:45)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:50)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter$1.transform(InProcessBuildActionExecuter.java:47)
[2019-10-23T16:53:15.396Z] 	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:78)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:47)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:31)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:42)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:28)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:78)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:52)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:59)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecuter.execute(SubscribableBuildActionExecuter.java:36)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:68)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SessionScopeBuildActionExecuter.execute(SessionScopeBuildActionExecuter.java:38)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:37)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:26)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:60)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:32)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:55)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:41)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:48)
[2019-10-23T16:53:15.396Z] 	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:32)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:68)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:27)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:78)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:75)
[2019-10-23T16:53:15.396Z] 	at org.gradle.util.Swapper.swap(Swapper.java:38)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:75)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:82)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
[2019-10-23T16:53:15.396Z] 	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[2019-10-23T16:53:15.396Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2019-10-23T16:53:15.396Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2019-10-23T16:53:15.396Z] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[2019-10-23T16:53:15.396Z] 	at java.base/java.lang.Thread.run(Thread.java:835)
[2019-10-23T16:53:15.396Z] Caused by: java.io.IOException: request to write '8192' bytes exceeds size in header of '6249522' bytes for entry 'docs/build/testclusters/integTest-0/logs/es.stdout.log'
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.tar.TarOutputStream.write(TarOutputStream.java:413)
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.ant.taskdefs.Tar.tarResource(Tar.java:517)
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.ant.taskdefs.Tar.tarFile(Tar.java:383)
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.ant.taskdefs.Tar.tar(Tar.java:709)
[2019-10-23T16:53:15.396Z] 	at org.apache.tools.ant.taskdefs.Tar.execute(Tar.java:347)
[2019-10-23T16:53:15.396Z] 	... 115 more
[2019-10-23T16:53:15.396Z] 
[2019-10-23T16:53:15.396Z] FAILURE: Build failed with an exception.
[2019-10-23T16:53:15.396Z] 
[2019-10-23T16:53:15.396Z] * What went wrong:
[2019-10-23T16:53:15.396Z] Execution failed for task ':x-pack:plugin:integTestRunner'.
[2019-10-23T16:53:15.396Z] > There were failing tests. See the report at: file:///var/jenkins_home/jobs/elasticsearch-pipeline/workspace/elasticsearch/x-pack/plugin/build/reports/tests/integTestRunner/index.html
[2019-10-23T16:53:15.396Z] 
[2019-10-23T16:53:15.396Z] * Try:
[2019-10-23T16:53:15.396Z] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[2019-10-23T16:53:15.396Z] 
[2019-10-23T16:53:15.396Z] * Get more help at https://help.gradle.org
[2019-10-23T16:53:15.396Z] 
[2019-10-23T16:53:15.396Z] BUILD FAILED in 16m 56s
[2019-10-23T16:53:15.396Z] 313 actionable tasks: 11 executed, 302 up-to-date
```

**Comment by atorok** [comment]
I don't think we have any restrictions on sizing ... anyhow this is no longer be considered a fatal error in master

**Pull request merged**